packages/custom_lint/test/src/workspace_test.dart

```diff
@@ -714,22 +738,38 @@ void main() {
         );
       }
 
-      test('throws MissingPubspecError if package does not contain a pubspec',
-          () async {
-        final workspace = await createSimpleWorkspace([]);
-        workspace.dir('package').createSync(recursive: true);
-
-        expect(
-          () => fromContextRootsFromPaths(
-            [p.join(workspace.path, 'package')],
+      test(
+        'finds pubspecs above analysis options file if there exists one',
+        () async {
+          final workspace = await createSimpleWorkspace(
+            [
+              Pubspec(
+                'plugin1',
+                version: Version(1, 0, 0),
+                dependencies: {'custom_lint_builder': HostedDependency()},
+              ),
+              Pubspec(
+                'a',
+                version: Version(1, 0, 0),
+                devDependencies: {
+                  'plugin1': HostedDependency(),
+                },
+              ),
+            ],
+            withNestedAnalysisOptions: true,
+          );
+          workspace.dir('package').createSync(recursive: true);
+          final customLintWorkspace = await CustomLintWorkspace.fromPaths(
+            [''],
```

Comment:
I'm skeptical about this empty string. That's not how `fromPaths` is used usually.

And I don't see folks every writing `custom_lint ""`

Reply:
I've raised an issue about the bug https://github.com/invertase/dart_custom_lint/issues/157

That bug is out of the scope of this PR. But your test should be updated